src/question-service/src/controllers/question-controller.js

```diff
@@ -0,0 +1,93 @@
+const Question = require("../models/question-model");
+const Counter = require("../models/counter-model");
+
+exports.createQuestion = async (req, res) => {
+  const { title, description, category, complexity } = req.body;
+  try {
+    const counter = await Counter.findOneAndUpdate(
```

Comment:
psps if i misunderstood but what is the counter model used for? is it used purely for the question id? or do we have features or use cases in future that will all utilise the counter. or r we just gonna count everything

Reply:
cuz if its just for the id and we dont have other use cases planned, we might want to use other means of getting an id. like using uuid. but not say is a lot better or anything uh is just i thought one less thing for us to take note off and maintain

Reply:
MongoDB documents have an auto-generated unique `_id` field which we can use.

But I thought having an integer id to identify questions would be more user-friendly (similar to Leetcode).

I think some SQL databases have an auto-incrementing feature, but this is a [workaround in Mongo](https://www.mongodb.com/basics/mongodb-auto-increment).

Reply:
Ohh okok i see but if we using this to display for ysers, then willthere be issues when we delete a question? All the ids will have to change right? If not when we browse all questions there will be some ids missing

Reply:
I was thinking that the `questionID` uniquely identifies questions and should be immutable.
We'll need it to guarantee an ordering of the questions too (if I'm not wrong, MongoDB queries don't guarantee any ordering). The alternative would be to use a creation timestamp.

I guess the missing IDs can't be helped, since updating every affected document in the DB will be quite expensive.

Reply:
ahh okok i get what u mean with the ordering thing